docs/rules/no-hooks-in-pages-folder.md

```diff
@@ -0,0 +1,58 @@
+# no-hooks-in-pages-folder
+
+Disallow hooks in Next.js pages folders
+
+## Examples
+
+👎 Examples of **incorrect** code for this rule
+
+```tsx
+// pages/index.tsx
+import { useQuery } from 'react-query';
+
+export default function Home() {
+	const { data } = useQuery('todos', () => fetch('/api/todos'));
+
+	return (
+		<div>
+			<h1>My Todos</h1>
+			<ul>
+				{data.map((todo) => (
+					<li key={todo.id}>{todo.title}</li>
+				))}
+			</ul>
+		</div>
+	);
+}
+```
+
+👍 Examples of **correct** code for this rule
+
+<!-- prettier-ignore -->
+```tsx
+// components/features/todos.ts
+import { useQuery } from 'react-query';
+
+export function TodosSection() {
+	const { data } = useQuery('todos', () => fetch('/api/todos'));
+
+	return (
+		<div>
+			<h1>My Todos</h1>
+			<ul>
+				{data.map((todo) => (
+					<li key={todo.id}>{todo.title}</li>
+				))}
+			</ul>
+		</div>
+	);
+}
+
+
+// pages/index.tsx
+import { TodosSection } from 'components/features/todos';
+
+export default function Home() {
+	return <TodosSection />;
+}
```

Comment:
```suggestion
export function Todos() {
	const { data } = useQuery('todos', () => fetch('/api/todos'));
	
	if (!data) {
	  return <div>Loading...</div>;
	}

	return (
			<ul>
				{data.map((todo) => (
					<li key={todo.id}>{todo.title}</li>
				))}
			</ul>
	);
}


// pages/index.tsx
import { Todos } from 'components/features/todos';

export default function Home() {
	return (
		<div>
			<h1>My Todos</h1>
			<ul>
				<Todos />
			</ul>
		</div>
	);
}
```
I don't like the section example because in this example Todos component is an island. At least I see it that way.
With this example it will be easier to wrap component with `<Suspense>`, also it demonstrates the importance of loading state in the early return.

Reply:
Changed [3f47d68](https://github.com/infinum/js-linters/pull/111/commits/3f47d68cf1c09fcaccc6a75c634c496a236a2a19)